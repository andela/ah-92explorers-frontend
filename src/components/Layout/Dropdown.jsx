/* eslint-disable jsx-a11y/no-static-element-interactions */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { logout } from '../../redux/actions/actionCreators/login';
import '../../assets/css/dropdown.css';

export class Dropdown extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displayMenu: false,
    };
  }

  showDropdownMenu = (event) => {
    event.preventDefault();
    this.setState({ displayMenu: true }, () => {
      document.addEventListener('click', this.hideDropdownMenu);
    });
  }

  hideDropdownMenu = () => {
    this.setState({ displayMenu: false }, () => {
      document.removeEventListener('click', this.hideDropdownMenu);
    });
  }

  logout = (e) => {
    e.preventDefault();
    this.props.logout();
  }

  render() {
    return (
      <div className="menu-dropdown" style={{ width: '100px' }}>
        <div className="user-caret" onClick={this.showDropdownMenu} onKeyDown={this.handleClick}>
          <span className="fa fa-caret-down" />
        </div>

        { this.state.displayMenu ? (
          <ul>
            <li><Link to="/profile">Profile</Link></li>
            <hr />
            <li><button type="button" onClick={this.logout} className="signout">Sign out</button></li>
          </ul>
        )
          : (
            null
          )}

      </div>

    );
  }
}

Dropdown.propTypes = {
  logout: PropTypes.func.isRequired,
};

function mapStateToProps(state) {
  return {
    login: state.login,
  };
}


export default connect(mapStateToProps, { logout })(Dropdown);
