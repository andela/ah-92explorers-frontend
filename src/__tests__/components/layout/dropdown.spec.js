import React from 'react';
import { shallow } from 'enzyme';
import { Dropdown } from '../../../components/Layout/Dropdown.jsx';

const props = {
  displayMenu: false,
  logout: jest.fn(),
  login: { user: { username: 'efefdf' }, isAuthenticated: true },
};
let wrapper;

describe('<Dropdown />', () => {
  beforeAll(() => {
    wrapper = shallow(<Dropdown {...props} />);
  });

  it('should render dropdown component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show dropdown menu', () => {
    const fakeEvent = { preventDefault: () => {} };
    const component = wrapper.instance();
    component.showDropdownMenu(fakeEvent);
    expect(component).toBeDefined();
  });

  it('should hide dropdown menu', () => {
    const component = wrapper.instance();
    component.hideDropdownMenu();
    expect(component).toBeDefined();
  });

  it('should hande logout button ', () => {
    const fakeEvent = { preventDefault: () => {} };
    const component = wrapper.instance();
    component.logout(fakeEvent);
    expect(component).toBeDefined();
  });
});
