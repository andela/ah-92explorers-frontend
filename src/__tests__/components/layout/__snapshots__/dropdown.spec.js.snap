// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should render dropdown component 1`] = `
<div
  className="menu-dropdown"
  style={
    Object {
      "width": "100px",
    }
  }
>
  <div
    className="user-caret"
    onClick={[Function]}
  >
    efefdf
     
    <span
      className="fa fa-caret-down"
    />
  </div>
</div>
`;
